int16 x
int16 y
float32 val
float32 time_sec